components/prize_pool/commons/prize_pool_placement_base.lua

```diff
@@ -164,7 +164,7 @@ function BasePlacement:parseOpponentArgs(input, date)
 	elseif type(opponentArgs[1]) ~= 'table' then
 		opponentData = Opponent.readOpponentArgs(opponentArgs)
 	end
-	---@cast opponentData standardOpponent
+	---@cast opponentData standardOpponent?
```

Comment:
Probably better setting `---@type standardOpponent?` on L161 instead of casting

Reply:
can't as we overwrite it in line 163: ``This variable is defined as type `standardOpponent?`. Cannot convert its type to `string|table`.``

if really wanted we could do an additional `--[[@as standardOpponent]]` in line 163

Reply:
Where is it being defined as a string? That's the problem. Casting should be avoid as much as possible

Reply:
it isn't
it even does a type check but it doesn't get it properly
![IMG_3788](https://github.com/Liquipedia/Lua-Modules/assets/75081997/136dd7bc-6aac-4559-a91e-5f97b6a50047)

Reply:
type at line 163 works though :P